### ISSUE TITLE
SQLServerResultSetMetaData performs checkClosed and that's causing issues

### ISSUE THREAD

**Issue by phaninra** [Microsoft/mssql-jdbc#678]
## Driver version or jar name
6.3.3.jre7-preview

## SQL Server version
2016

## Client operating system
Windows 7

## Java/JVM version
"1.8.0"

## Problem description
The following example can be found in the javadocs of java.sql.ResultSetMetaData

```
ResultSet rs = stmt.executeQuery("SELECT a, b, c FROM TABLE2");
ResultSetMetaData rsmd = rs.getMetaData();
int numberOfColumns = rsmd.getColumnCount();
```

I've a similar operation which uses `org.springframework.jdbc.object.MappingSqlQuery<T>` to execute the query and then gets hold of ResultSetMetaData object.

The call `getColumnCount` works fine with jTDS but not with `mssql` driver as `SQLServerResultSetMetaData` checks whether the connection is closed.

I might be missing something but could not figure it out. Thanks.

**Comment by ulvii** [comment]
Hi @phaninra ,

Just to confirm, does the issue happen when the result set is closed before `rsmd.getColumnCount();`?

So repro code would be something like this?
```
ResultSet rs = stmt.executeQuery("SELECT a, b, c FROM TABLE2");
ResultSetMetaData rsmd = rs.getMetaData();
rs.close()
int numberOfColumns = rsmd.getColumnCount();
```

**Comment by phaninra** [comment]
Yep, and I believe it's due to this - https://github.com/spring-projects/spring-framework/blob/master/spring-jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java#L670 . 

I realize that it's Spring JdbcTemplate which is closing the ResultSet. However, is there any reason for checking if the connection has been closed inside `SQLServerResultSetMetaData`?

**Comment by ulvii** [comment]
Fixed in #685
